addons/mail/static/src/new/web/messaging_menu/notification_item.js

```diff
@@ -3,9 +3,11 @@
 import { Component, useRef } from "@odoo/owl";
 import { ImStatus } from "@mail/new/discuss/im_status";
 import { RelativeTime } from "@mail/new/core_ui/relative_time";
+import { ActionSwiper } from "@web/core/action_swiper/action_swiper";
+import { isMobileOS } from "@web/core/browser/feature_detection";
```

Comment:
I just realized we also have `hasTouch` helper, which is even better for this feature.

But I'm wondering, do we really need to make that check after all? It would make more sense if the Swiper component was directly handling it (or if nothing happened if touch is not available either way)?

Reply:
Aren't both working? Similar to how it's possible to touch or click in mobile?

Reply:
Changed (uses `hasTouch`)

Reply:
Hello, good to see the feature reimplemented!
From my experience, `hasTouch` don't recognize laptops with touchscreen. Currently, the feature is usable on those devices when the screen is smaller (`env.isSmall`). I think it's not that bad to support the feature also for laptops and Windows tablets, which unfortunately doesn't work with `hasTouch`. But you're free to drop the support of course :)